python/cm/upgrade.py

```diff
@@ -256,6 +256,24 @@ def rpm_cmp_reverse(obj1, obj2):
         return res
 
 
+def recount_issues(obj, upgrade):
+    """
+    Recount of issues from the bottom up for all related objects component-service-cluster
+    or host-provider, or for cluster or provider objects
```

Comment:
>  or for cluster or provider objects

Incorrect. According to code you do recursion for cluster and provider and do no recursion for component, service and host.

Reply:
Fixed